contracts/modules/SwapsImplSovrynSwapModule.sol

```diff
@@ -0,0 +1,109 @@
+        _setTarget(this.getSovrynSwapNetworkContract.selector, target);
+        _setTarget(this.getContractHexName.selector, target);
+        _setTarget(this.swapsImplInternalExpectedRate.selector, target);
+        _setTarget(this.swapsImplInternalExpectedReturn.selector, target);
+        emit ProtocolModuleContractReplaced(
+            prevModuleContractAddress,
+            target,
+            "SwapsImplSovrynSwapModule"
+        );
+    }
+
+    /**
+     * Get the hex name of a contract.
+     * @param source The name of the contract.
+     * */
+    function getContractHexName(string memory source) public pure returns (bytes32 result) {
+        return _getContractHexName(source);
+    }
+
+    /**
+     * Look up the Sovryn swap network contract registered at the given address.
+     * @param sovrynSwapRegistryAddress The address of the registry.
+     * */
+    function getSovrynSwapNetworkContract(address sovrynSwapRegistryAddress)
+        public
+        view
+        returns (ISovrynSwapNetwork)
+    {
+        return _getSovrynSwapNetworkContract(sovrynSwapRegistryAddress);
+    }
+
+    /**
+     * @notice Get the expected rate for 1 source token when exchanging the
+     *   given amount of source tokens.
+     *
+     * @param sourceTokenAddress The address of the source token contract.
+     * @param destTokenAddress The address of the destination token contract.
+     * @param sourceTokenAmount The amount of source tokens to get the rate for.
+     * */
+    function swapsImplInternalExpectedRate(
+        address sourceTokenAddress,
+        address destTokenAddress,
+        uint256 sourceTokenAmount,
+        address sovrynSwapContractRegistryAddress
+    ) external view returns (uint256) {
+        return
+            internalExpectedRate(
+                sourceTokenAddress,
+                destTokenAddress,
+                sourceTokenAmount,
+                sovrynSwapContractRegistryAddress
+            );
+    }
+
+    /**
+     * @notice Get the expected return amount when exchanging the given
+     *   amount of source tokens.
+     *
+     * @notice Right now, this function is being called directly by _swapsExpectedReturn from the protocol
+     * So, this function is not using getConversionPath function since it will try to read the defaultPath storage which is stored in the protocol's slot, and it will cause an issue for direct call.
+     * Instead, this function is accepting additional parameters called defaultPath which value can be declared by the caller (protocol in this case).
+     *
+     * @param sourceTokenAddress The address of the source token contract.
+     * @param destTokenAddress The address of the destination token contract.
+     * @param sourceTokenAmount The amount of source tokens to get the return for.
+     * @param sovrynSwapContractRegistry The sovryn swap contract reigstry address.
+     * */
+    function swapsImplInternalExpectedReturn(
```

Comment:
```suggestion
    function swapsImplExpectedReturn(
```

Reply:
updated in this [commit](https://github.com/DistributedCollective/Sovryn-smart-contracts/pull/530/commits/0334bfa7656f28d673a27380fd05475bcc0b6524)